routes/users.js

```diff
@@ -1,18 +1,74 @@
 const express = require("express");
 const router = express.Router();
 const jwt = require("jsonwebtoken");
+const User = require("../models/User");
+const authenticateToken = require("../middlewares/authToken");
+// const bcrypt = require("bcryptjs");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
 
-router.post(/login/, async (req, res) => {});
+router.post("/register", async (req, res) => {
+    const { username, password, role } = req.body;
 
-function authenticateToken(req, res, next) {
-    const authHeader = req.headers["authorization"];
-    const token = authHeader && authHeader.split(" ")[1];
-    if (token == null) return res.sendStatus(401);
+    if (!username || !password) {
+        return res.status(400).json({ message: "Missing username or password" });
+    }
 
-    jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
-        console.log(err);
-        if (err) return res.sendStatus(403);
-        req.user = user;
-        next();
+    const user = await User.findOne({ username });
+    if (user) {
+        return res.status(400).json({ message: "User already exists" });
+    }
+
+    hash = bcrypt.hash(password, saltRounds, async function (err, hash) {
+        if (err) return res.status(500).json({ message: "Error hashing password" });
```

Comment:
The messages on the errors need to be more generic. You are exposing details about your flow that can be used by hackers to retrieve information about your site.